src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -107,6 +115,57 @@ public void register(Router router) {
           routingContext, webContext));
   }
 
+  private void move(RoutingContext routingContext) {
+
+    WebContext context = new WebContext(routingContext);
+    JsonObject itemsMoveJsonRequest = routingContext.getBodyAsJson();
+
+    Optional<ValidationError> validationError = itemsMoveHasRequiredFields(itemsMoveJsonRequest);
+    if (validationError.isPresent()) {
+      unprocessableEntity(routingContext.response(), validationError.get());
+    } else {
+      ItemsMove itemsMove = MoveUtil.jsonToItemsMove(itemsMoveJsonRequest);
+      List<String> itemIdsToUpdate = itemsMove.getItemIds();
+      try {
+        storage.getItemCollection(context).findByCql(
+          buildQueryByIds(itemIdsToUpdate),
+          PagingParameters.defaults(),
+          success -> updateManyItemsWithNewHoldingsRecordId(routingContext, context, success.getResult(), itemsMove.getToHoldingsRecordId(), itemIdsToUpdate),
+          FailureResponseConsumer.serverError(routingContext.response()));
+      } catch (UnsupportedEncodingException e) {
+        ServerErrorResponse.internalError(routingContext.response(), e.toString());
+      }
+    }
+  }
+
+  private void updateManyItemsWithNewHoldingsRecordId(
+    RoutingContext routingContext,
+    WebContext context,
+    MultipleRecords<Item> itemsToUpdate,
+    String toHoldingsRecordId,
+    List<String> itemIdsToUpdate) {
+
+    ItemCollection storageItemCollection = storage.getItemCollection(context);
+
+    List<CompletableFuture<Item>> updates = itemsToUpdate.records.stream()
```

Comment:
For cross instance updates we need to notify mod-circulation about it, because there are some places where item data is duplicated and stored in mod-circulation-storage (e.g. requests store identifiers).

Reply:
@bohdan-suprun 
Does using of PUT inventory/items/ instead of items-storage/items/ cover mod-circulation notifications? I mean if I use items updating on inventory (not items-storage directly) then we don't need to implement additional logic for circulation notifications?

Reply:
At the moment, that information is not kept in sync until the request itself is updated. This is a known significant limitation.

I don't think we should try to extend that as part of this work. 

@bohdan-suprun does that make sense?

Reply:
@khandramai, @marcjohnson-kint sure, I think it is out of scope to sync circulation. It worth letting POs know that we have this challenge.